src/reflect/type.go

```diff
@@ -1284,3 +1286,25 @@ func StructOf([]StructField) Type {
 func MapOf(key, value Type) Type {
 	panic("unimplemented: reflect.MapOf()")
 }
+
+const maxVarintLen32 = 5
+
+// encoding/binary.Uvarint, specialized for uint32
+func uvarint32(buf []byte) (uint32, int) {
+	var x uint32
+	var s uint
+	for i, b := range buf {
+		if i == maxVarintLen32 {
+			return 0, -(i + 1) // overflow
+		}
+		if b < 0x80 {
+			if i == maxVarintLen32-1 && b > 1 {
+				return 0, -(i + 1) // overflow
+			}
+			return x | uint32(b)<<s, i + 1
+		}
```

Comment:
We know that the input data is correct, so we don't need to check for overflow. I suggest removing those checks to reduce binary size.